packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -58,33 +61,38 @@ library V1Proposing {
     }
 
     function proposeBlock(
-        LibData.State storage state,
-        LibData.TentativeState storage tentative,
+        TaikoData.State storage state,
+        TaikoData.Config memory config,
+        TaikoData.TentativeState storage tentative,
         AddressResolver resolver,
         bytes[] calldata inputs
     ) public onlyWhitelistedProposer(tentative) {
-        assert(!V1Utils.isHalted(state));
+        assert(!LibUtils.isHalted(state));
 
         require(inputs.length == 2, "L1:inputs:size");
-        LibData.BlockMetadata memory meta = abi.decode(
+        TaikoData.BlockMetadata memory meta = abi.decode(
             inputs[0],
-            (LibData.BlockMetadata)
+            (TaikoData.BlockMetadata)
         );
-        _verifyBlockCommit(state, meta);
-        _validateMetadata(meta);
+        _verifyBlockCommit({
+            state: state,
+            commitConfirmations: config.commitConfirmations,
+            meta: meta
+        });
+        _validateMetadata(config, meta);
 
         {
             bytes calldata txList = inputs[1];
             // perform validation and populate some fields
             require(
-                txList.length >= 0 &&
-                    txList.length <= LibConstants.K_TXLIST_MAX_BYTES &&
+                txList.length > 0 &&
```

Comment:
Seems we will not allow proposing empty block by giving empty encoded txlist bytes? Is this change intended?

Reply:
Good catch, fixed.